### PR TITLE
Implement memory tracker layer for GAPID/Perfetto

### PR DESCRIPTION
**Pull request by zakerinasab**
This is the GAPID side of the code for a memory tracker layer
added to GAPID profiler. The Perfetto side of the code is already
landed in Perfetto repo.

Bug: b/138664961

### CODE REVIEW

File: core/vulkan/perfetto_producer/perfetto_threadlocal_emitter.inc [hunk 1]
```diff
@@ -18,180 +18,226 @@
...
-        if (!*interned_data) {
-            *interned_data = packet->set_interned_data();
-        }
-        auto* event_name = (*interned_data)->add_debug_annotation_names();
-        event_name->set_iid(val);
-        event_name->set_name(name);
-        return interned_annotation_names_.count();
+    const char *name,
+    typename PerfettoProducer<T>::TraceContext::TracePacketHandle &packet,
+    perfetto::protos::pbzero::InternedData **interned_data) {
```
Comment (AWoloszyn) [inline core/vulkan/perfetto_producer/perfetto_threadlocal_emitter.inc]: Something bad happened to formatting, specifically *

Comment (zakerinasab) [reply core/vulkan/perfetto_producer/perfetto_threadlocal_emitter.inc]: This file is clang-formatted now, may be that's why.

Comment (AWoloszyn) [reply core/vulkan/perfetto_producer/perfetto_threadlocal_emitter.inc]: It looks to be the wrong clang-formatting rules however?

Comment (zakerinasab) [reply core/vulkan/perfetto_producer/perfetto_threadlocal_emitter.inc]: Strange. Not sure how this happened. Fixed it.

**Comment by AWoloszyn** [comment]
Once you rebase this on top of master this is probably good to go.

**Comment by lpy** [comment]
The `cmd/gapit/BUILD.bazel` change is causing `bazel run gapit` failed on mac

**Comment by zakerinasab** [comment]
This is now fixed after https://github.com/google/gapid/pull/3510 is merged.

**Pull request merged**